README.md

```diff
@@ -35,49 +36,81 @@ npm i @libp2p/webrtc
-const { stream } = await node.dialProtocol(addr, '/my-protocol/1.0.0')
-const values = await pipe(stream, all)
+import first from "it-first";
+import { pipe } from "it-pipe";
+import { fromString, toString } from "uint8arrays";
+import { webRTC } from 'js-libp2p-webrtc'
+
+const node = await createLibp2p({
+  transports: [webRTC()],
+  connectionEncryption: [() => new Noise()],
+});
+
+await node.start()
+
+const ma =  multiaddr('/ip4/0.0.0.0/udp/56093/webrtc/certhash/uEiByaEfNSLBexWBNFZy_QB1vAKEj7JAXDizRs4_SnTflsQ')
+const stream = await node.dialProtocol(ma, ['/my-protocol/1.0.0']) 
+const message = `Hello js-libp2p-webrtc\n`
+const response = await pipe([fromString(message)], stream, async (source) => await first(source))
+const responseDecoded = toString(response.slice(0, response.length))
 ```
-## API
+
+## Examples
+Examples can be found in the [examples folder](examples/README.md).
+
+## Interfaces
 
 ### Transport
 
-[![](https://raw.githubusercontent.com/libp2p/js-libp2p-interfaces/master/packages/libp2p-interfaces/src/transport/img/badge.png)](https://github.com/libp2p/js-libp2p-interfaces/tree/master/packages/libp2p-interfaces/src/transport)
+![https://github.com/libp2p/js-libp2p-interfaces/tree/master/packages/interface-transport](https://raw.githubusercontent.com/libp2p/js-libp2p-interfaces/master/packages/interface-transport/img/badge.png)
 
-`libp2p-webrtc` accepts WebRTC encapsulated addresses: `/ip4/0.0.0.0/udp/56093/webrtc/certhash/uEiByaEfNSLBexWBNFZy_QB1vAKEj7JAXDizRs4_SnTflsQ`
+Browsers can only `dial`, so `listen` is not supported.
 
-### Connection
+```js
+interface Transport {
+  [Symbol.toStringTag]: string
+  [symbol]: true
+  dial: (ma: Multiaddr, options: DialOptions) => Promise<Connection>
+  createListener: (options: CreateListenerOptions) => Listener
+  filter: MultiaddrFilter
+}
+
+class WebRTCTransport implements Transport {
+
+  async dial (ma: Multiaddr, options: WebRTCDialOptions): Promise<Connection> {
+    const rawConn = await this._connect(ma, options)
+    log(`dialing address - ${ma.toString()}`)
+    return rawConn
+  }
+
+  createListener (options: CreateListenerOptions): Listener {
+    throw unimplemented('WebRTCTransport.createListener')
+  }
+}
+```
 
-[![](https://raw.githubusercontent.com/libp2p/js-libp2p-interfaces/master/packages/libp2p-interfaces/src/connection/img/badge.png)](https://github.com/libp2p/js-libp2p-interfaces/tree/master/packages/libp2p-interfaces/src/connection)
+### Connection
 
-## Hacking
+![https://github.com/libp2p/js-libp2p-interfaces/tree/master/packages/interface-connection](https://raw.githubusercontent.com/libp2p/js-libp2p-interfaces/master/packages/interface-connection/img/badge.png)
 
-Besides the usual `npm install` to get dependencies, `npm run build` to invoke tsc, and `npm run test` to execute unit tests...
+```js
+interface MultiaddrConnection extends Duplex<Uint8Array> {
+  close: (err?: Error) => Promise<void>
+  remoteAddr: Multiaddr
```

Comment:
Does it has to be writable or can it be `readonly`? Same question for the next property.